plugins/poll/lib/poll.rb

```diff
@@ -306,6 +306,10 @@ def self.create!(post_id, poll)
   end
 
   def self.extract(raw, topic_id, user_id = nil)
+    # Poll Post handlers get called very early in the post
```

Comment:
`raw` is one of the things I expected to be present from the time the Post object is initialized. Can you elaborate?

Reply:
Plugins are able to hook into  the new post creation process by registering handlers  with `NewPostManger`. These registered handlers are called long before any occurrence of Post object instantiation and validation in core.

https://github.com/discourse/discourse/blob/73ec80893d750e82d0cdaf3f57383751566031d7/lib/new_post_manager.rb#L214-L245

With the exception of parameter validations in the `PostsController` and the `WordWatcher` validations shown above, none of the Post specific validation run before the `extract` method call in the poll plugin.